app/src/main/java/com/zulip/android/networking/AsyncGetEvents.java

```diff
@@ -190,7 +190,13 @@ public Void call() throws Exception {
                             } else {
                                 personDao.update(person);
                             }
-                        } catch (Exception e) {
+                            if (person.getEmail().equals(app.getEmail())) {
```

Comment:
@vishwesh3 yes, I agree, the name and avatarUrl would be null immediately after reset. In that case I think it'd be reasonable to check for email equality `if (person.getEmail().equals(app.getYou.getEmail()))` in `narrow_pm_with()`. If we go with this, I think we won't need this change in asyncGetEvents.java as the Person list would always contain 1 object referring to the user itself in case of narrow to self.